src/components/activity/ActivityRegistrator.tsx

```diff
@@ -10,36 +10,101 @@ import { ChoiceBasedTextInput } from '../ChoiceBasedTextInput';
 
 import { useTranslation } from 'language/LanguageProvider';
 import Storage from 'storage';
-import { ActivitiesEntry } from 'storage/types';
+import { ActivitiesDay, ActivitiesEntry } from 'storage/types';
 
 import { ISODate } from 'utils';
+import { getDaysInMonth } from 'react-native-paper-dates/lib/typescript/src/Date/dateUtils';
 
 export const ActivityRegistrator = ({ route, navigation }: any) => {
   const lang = useTranslation();
   const [settings, modifySettings] = Storage.useSettings();
   const { iconSizes, colors } = useTheme();
+  const [values, modifyValues] = Storage.useValues();
+  const [activities, modifyActivities] = Storage.useActivities();
 
   const steps = 1;
+  let dateDefault = new Date();
+  let fromTimeRoundedDefault = getCurrentTimeRounded(0, steps);
+  let toTimeRoundedDefault = getCurrentTimeRounded(1, steps);
+  let activityTextDefault = '';
+  let importanceDefault = 5;
+  let enjoymentDefault = 5;
 
-  const [values, modifyValues] = Storage.useValues();
-  const [activities, modifyActivities] = Storage.useActivities();
+  // Overwrite default values from entry (if it exists)
+  if (route.params?.entry && route.params?.day) {
```

Comment:
If route.params.entry is a index in the list much of this logic could be simplified. That would require the history view to be changed slightly, but not much (see that comment).

Reply:
I will look into this. I did not really think about this since my intentions at first was only to add the missing functionality. I took a look at calendar and saw how it was done there, will try to adapt a similar structure here.